addons/account/tests/common.py

```diff
@@ -985,6 +832,169 @@ def setup_armageddon_tax(cls, tax_name, company_data):
+                    'value_amount': 30.0,
+                    'sequence': 400,
+                    'days': 0,
+                    'option': 'day_after_invoice_date',
+                }),
+                (0, 0, {
+                    'value': 'balance',
+                    'value_amount': 0.0,
+                    'sequence': 500,
+                    'days': 31,
+                    'option': 'day_following_month',
+                }),
+            ],
+        })
+
+        # ==== Partners ====
+        obj.partner_a = obj.env['res.partner'].create({
+            'name': 'partner_a',
+            'property_payment_term_id': obj.pay_terms_a.id,
+            'property_supplier_payment_term_id': obj.pay_terms_a.id,
+            'property_account_receivable_id': obj.company_data['default_account_receivable'].id,
+            'property_account_payable_id': obj.company_data['default_account_payable'].id,
+            'company_id': False,
+        })
+        obj.partner_b = obj.env['res.partner'].create({
+            'name': 'partner_b',
+            'property_payment_term_id': obj.pay_terms_b.id,
+            'property_supplier_payment_term_id': obj.pay_terms_b.id,
+            'property_account_position_id': obj.fiscal_pos_a.id,
+            'property_account_receivable_id': obj.company_data['default_account_receivable'].copy().id,
+            'property_account_payable_id': obj.company_data['default_account_payable'].copy().id,
+            'company_id': False,
+        })
+
+        # ==== Cash rounding ====
+        obj.cash_rounding_a = obj.env['account.cash.rounding'].create({
+            'name': 'add_invoice_line',
+            'rounding': 0.05,
+            'strategy': 'add_invoice_line',
+            'profit_account_id': obj.company_data['default_account_revenue'].copy().id,
+            'loss_account_id': obj.company_data['default_account_expense'].copy().id,
+            'rounding_method': 'UP',
+        })
+        obj.cash_rounding_b = obj.env['account.cash.rounding'].create({
+            'name': 'biggest_tax',
+            'rounding': 0.05,
+            'strategy': 'biggest_tax',
+            'rounding_method': 'DOWN',
+        })
+
+    # -------------------------------------------------------------------------
+    # HELPERS
+    # -------------------------------------------------------------------------
+
+    @classmethod
+    def copy_account(cls, account):
+        ''' Helper to copy an account without risking a collision regarding their code.
+        :param account: The account.account record to copy.
+        :return:        The newly copied account.account record.
+        '''
+        suffix_nb = 1
+        while True:
+            new_code = '%s (%s)' % (account.code, suffix_nb)
+            if account.search_count([('company_id', '=', account.company_id.id), ('code', '=', new_code)]):
+                suffix_nb += 1
+            else:
+                return account.copy(default={'code': new_code})
+
```

Comment:
mocked_today should be removed, maybe it's done in another branch?

Reply:
need to rebase